main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+    else
+        std::cout << "This passenger cannot board the elevator" << std::endl;
+}
+
+void Elevator::offboardPassenger()
+{
+    if(arrived() && lastUserWeight > 0)
+        updateWeightCarried(lastUserWeight, false);
+    else
+        std::cout << "The elevator has not arrived to a floor yet or no passengers inside" << std::endl;
+}
+
+void Elevator::updateWeightCarried(double userWeight, bool goingInside)
+{
+    if(goingInside)
+    {
+        lastUserWeight = userWeight;
+        totalWeightCarried += lastUserWeight;
+        std::cout << "Passenger onboarded" << std::endl;
+    }
+    else
+    {
+        totalWeightCarried -= lastUserWeight;
+        std::cout << "Passenger offboarded" << std::endl;
+    }
+}
+
+bool Elevator::canTakePassengerWeight(double userWeight)
+{
+    if((userWeight + totalWeightCarried) > maxWeight)
+        return false;
+        
+    return true;
+}
+
+bool Elevator::arrived()
+{
+    return elevatorArrived;
+}
+
+// ============= SchoolBag Functions ===============
+
+SchoolBag::SchoolBag()
+{
+    std::cout << "****** SchoolBag Created ******" << std::endl;
+}
+
+SchoolBag::~SchoolBag()
+{
+    unpackPencilCase();
+    std::cout << "****** SchoolBag Destroyed ******" << std::endl;
+}
+
+void SchoolBag::packPencilCase()
+{
+    if(!schoolPencilCase.inSchoolBag && !schoolPencilCase.isOpen)
+    {
+        schoolPencilCase.inSchoolBag = true;
+        std::cout << "Packing the pencilcase... Good to go!" << std::endl;
+    }
+    else if(schoolPencilCase.inSchoolBag)
+        std::cout << "Pencilcase already packed" << std::endl;
+    else
+    {
+        schoolPencilCase.removeItems(schoolPencilCase.maxItemsToHold);
+        std::cout << "** You forgot to close your pencilcase!! **" << std::endl;
+    }
+}
+
+void SchoolBag::unpackPencilCase()
+{
+    if(!schoolPencilCase.inSchoolBag)
+        std::cout << "Pencilcase already unpacked or not in schoolbag" << std::endl;
+    else
+    { 
+        schoolPencilCase.inSchoolBag = false;
+        std::cout << "Unpacked the pencilcase" << std::endl;
+    }
+}
+
+// ============= Building Functions ==============
+
+Building::Building()
+{
+    std::cout << "****** Building Created ******" << std::endl;
+}
+
+Building::~Building()
+{
+    closeBuilding();
+    std::cout << "****** Building Destroyed ******" << std::endl;
+}
+
+void Building::closeBuilding()
+{
+    if(buildingEmpty() && isOpen)
+    {
+        isOpen = false;
+        std::cout << "Closing building. Elevators clear..." << std::endl;
+    }
+    else if(!isOpen)
+        std::cout << "Building already closed" << std::endl;
+    else
+        std::cout << "Cannot close building now.. There are still users in the elevator" << std::endl;
+}
+
+void Building::openBuilding()
+{
+    if(!isOpen)
+    {
+        isOpen = true;
+        std::cout << "Building is now open" << std::endl;
+    }
+    else
+        std::cout << "Building is already open" << std::endl;
+}
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise